### DISCUSSION TITLE
Firmware upload works & device is found. Everything else is not

### DISCUSSION THREAD

**Discussion by HausBJB** [nanoframework/Home#1186] [Q&A]
### Target name(s)

ESP32_BLE_REV0

### Firmware version

1.8.0.616

### Was working before? On which version?

N/A

### Device capabilities

nanoff --update --target ESP32_BLE_REV0 --serialport COM10
.NET nanoFramework Firmware Flasher v2.4.2+c4df2f6716
Copyright (C) 2019 .NET Foundation and nanoFramework project contributors

Using COM10 @ 1500000 baud to connect to ESP32.
Reading details from chip...
OK

Connected to:
ESP32 (ESP32-D0WDQ6 (revision 1))
Features WiFi, BT, Dual Core, 240MHz, VRef calibration in efuse, Coding Scheme None
Flash size 4MB unknown from TENX (manufacturer 0x94 device 0x16406)
PSRAM: undetermined
Crystal 40MHz
MAC C8:F0:9E:A3:18:E4

Trying to find ESP32_BLE_REV0 in stable repository...OK
Extracting ESP32_BLE_REV0-1.8.0.616.zip...OK

Updating to 1.8.0.616

Erasing flash...
OK
Flashing firmware...
Wrote 23856 bytes (15078 compressed) at 0x00001000 in 0.4 seconds (effective 477.1 kbit/s)...
Hash of data verified.
Wrote 1254352 bytes (823172 compressed) at 0x00010000 in 12.9 seconds (effective 778.9 kbit/s)...
Hash of data verified.
Wrote 3072 bytes (136 compressed) at 0x00008000 in 0.1 seconds (effective 305.0 kbit/s)...
Hash of data verified.
OK

### Description

The firmware update worked wonderfully. VS 2022 also finds the device in the device explorer.
Unfortunately, all other things do not work.

First start:
17:28:25.755 [NanoDevices: checking device  COM10 *** 2nd attempt ***]
17:28:25.779 [NanoDevices: checking device  COM10 @ 921600]
17:28:29.478 [NanoDevices: valid device ESP32 @ COM10]
17:28:29.478 [NanoDevices: Serial device enumeration completed. Found 1 devices]

E.g. Ping works and I get the following message.

ESP32 @ COM10 is active running nanoCLR.

All other commands do not work e.g
Device capabilities report the following.

ESP32 @ COM10 is not responding, please reboot the device.

F5 =>
17:34:48.783 [Starting deployment transaction from v2022.2.0.48+5f8dab6e5f]
17:34:48.786 [Starting debug engine on nanoDevice]
17:34:48.812 [Connecting to debugger engine...]
17:34:53.200 [*** ERROR: failing to connect to device ***]

Ping
ESP32 @ COM10 is active running nanoCLR.

Device Capabilities
ESP32 @ COM10 is not responding, please reboot the device.

What am I doing wrong?

### Screenshots

![image](https://user-images.githubusercontent.com/53478848/202238353-aae95119-592e-44ed-97a4-85316d8f7776.png)
![image](https://user-images.githubusercontent.com/53478848/202240666-928fe866-10ca-478d-97d2-e36eef1d6b40.png)

![photo_2022-11-16_17-43-44](https://user-images.githubusercontent.com/53478848/202241305-9c76907a-e1b2-4909-b2dd-7cf29584d2ae.jpg)

**Comment by josesimoes** [comment]
Hey @HausBJB! As stated in the issues template, we're using github issue to track work items, not anwesering questions.
I'm moving this to discussions. 
Nevertheless, I suggest that you join our [Discord server](https://discordapp.com/invite/gCyBu8T) for quick and interactive support.

**Comment by HausBJB** [comment]
OK thanks. @josesimoes

But couldn't that also be a bug?

Then I hope that someone can help me here

**Reply by alberk8** [reply]
You could have a lemon here.  The inscription on the metal lid of the ESP32 says that it is a ESP32-WROVER-E which mean that is is a REV3 MCU, where as the flashing tool says that it is a REV1 MCU.  So it would be hard to trouble shoot.  Most like you have a third rate or reject MCU.  I have encountered a few from buying from sites offering for cheap :(.

**Reply by HausBJB** [reply]
hi yes i saw that too. Unfortunately the Wrover firmware does not work. I have the PIco on it and can also play programs on it. Unfortunately, the PiCo does not have BLE support

**Comment by MexHigh** [comment]
I have the same problem with an ESP WROOM 32. Flashing the Program succeeds in 1 out of 20 deployment tries. On the other tries I get the same "ERROR: failing to connect to device". I tried multiple targets (ESP32_REV0, ESP32_REV3 (and BLE versions), ESP32_WROOM_32). 

Don't know what I'm doing wrong. I checked the cable multiple times and tried with multiple ESPs from the same model.

**Reply by josesimoes** [reply]
That sounds like a power issue... Deploying involves writing to flash which requires proper power supply. Power glitches, faulty capacitors, bad regulators, etc. all that can make a deployment fail. 1 out of 20 is not acceptable. Deployment is a very robust and has been vastly improved in the past. For a long time it hasn't had any complaints or failure reports.

**Reply by alberk8** [reply]
Or it could be the LDO on the ESP32 dev board.  I have one that works (can power on and run app) but problematic when deploying.  When I replace the LDO, it works without issue.

**Comment by 0use-TE** [comment]
@HausBJB Hello, I have the same problem as you. I have successfully solved this problem, but I don't know if this can help you. Wish you a happy life!
https://discord.com/channels/478725473862549535/1168637329368416386/threads/1378761361063940178
